Raven.Database/Bundles/MoreLikeThis/MoreLikeThis.cs

```diff
@@ -802,7 +803,20 @@ internal PriorityQueue<object[]> RetrieveTerms(RavenJObject jsonDocument)
 
                 if (value == null) continue;
 
-                AddTermFrequencies(new StringReader(value.ToString()), words, fieldName);
+                switch (value.Type)
+                {
+                    case JTokenType.Array:
+                        foreach (var item in (RavenJArray)value)
+                        {
+                            if (item.Type != JTokenType.String) break;
+                            AddTermFrequencies(new StringReader(item.Value<string>()), words, fieldName);
+                        }
+                        break;
+                    case JTokenType.String:
+                        AddTermFrequencies(new StringReader(value.Value<string>()), words, fieldName);
+                        break;
+                    default: continue;
```

Comment:
In this case I think we should throw rather than log.
We should throw an InvalidOperationException that the query is not supported with object field.